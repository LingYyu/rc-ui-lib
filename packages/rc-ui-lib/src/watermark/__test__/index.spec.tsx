import { render, cleanup } from '@testing-library/react';
import { setupJestCanvasMock } from 'jest-canvas-mock';
import React from 'react';
import { Watermark } from '..';

describe('Watermark', () => {
  let spyConsole: jest.SpyInstance;
  const mockRevoke = jest.fn();

  beforeEach(() => {
    spyConsole = jest.spyOn(console, 'error');
    spyConsole.mockImplementation(() => {
      return null;
    });
    global.URL.createObjectURL = jest.fn(() => 'run to here');

    setupJestCanvasMock();
  });
  afterEach(() => {
    cleanup();
    spyConsole.mockRestore();
    jest.restoreAllMocks();
    window.URL.revokeObjectURL = mockRevoke;
  });

  it('should render content', () => {
    const { container } = render(<Watermark content="rc-ui-lib" />);

    expect(container).toMatchSnapshot();
  });

  it('should render image', () => {
    Object.defineProperty(Image.prototype, 'src', {
      set() {
        this.onload();
      },
    });

    const { container } = render(
      <Watermark
        image="https://rancui.github.io/rc-ui-lib/rc-ui-lib.png"
        width={180}
        height={90}
      />,
    );

    expect(container).toMatchSnapshot();
  });
  it('should render image and test opacity', () => {
    Object.defineProperty(Image.prototype, 'src', {
      set() {
        this.onload();
      },
    });

    const { container } = render(
      <Watermark
        image="https://rancui.github.io/rc-ui-lib/rc-ui-lib.png"
        width={180}
        height={90}
        opacity={0.2}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render html', () => {
    const { container } = render(
      <Watermark width={150}>
        <div style={{ background: 'linear-gradient(45deg, #000 0, #000 50%, #fff 50%)' }}>
          <p style={{ mixBlendMode: 'difference', color: '#fff' }}>rc watermark</p>
        </div>
      </Watermark>,
    );

    expect(container).toMatchSnapshot();
  });

  it('test width, height, rotate, zIndex', () => {
    const { container } = render(
      <Watermark content="rc-ui-lib" width={180} height={90} rotate="22" zIndex={999} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('test false value fullPage', () => {
    const { container } = render(<Watermark content="rc-ui-lib" fullPage opacity={0.2} />);

    expect(container).toMatchSnapshot();
  });
});
