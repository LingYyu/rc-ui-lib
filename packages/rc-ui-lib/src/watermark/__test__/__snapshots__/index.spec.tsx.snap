// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watermark should render content 1`] = `
<div>
  <div
    class="rc-water-mark"
    style="z-index: 100; background-image: url(); background-size: 100px 100px;"
  >
    <div>
      <svg
        height="100"
        style="padding: 0px 0px 0px 0px; display: none;"
        viewBox="0 0 100 100"
        width="100"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <foreignobject
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            style="transform-origin: center; transform: rotate(-22deg);"
            xmlns="http://www.w3.org/1999/xhtml"
          >
            <span
              style="color: rgb(220, 222, 224);"
            >
              rc-ui-lib
            </span>
          </div>
        </foreignobject>
      </svg>
    </div>
  </div>
</div>
`;

exports[`Watermark should render html 1`] = `
<div>
  <div
    class="rc-water-mark"
    style="z-index: 100; background-image: url(); background-size: 150px 100px;"
  >
    <div>
      <svg
        height="100"
        style="padding: 0px 0px 0px 0px; display: none;"
        viewBox="0 0 150 100"
        width="150"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <foreignobject
          height="100"
          width="150"
          x="0"
          y="0"
        >
          <div
            style="transform-origin: center; transform: rotate(-22deg);"
            xmlns="http://www.w3.org/1999/xhtml"
          >
            <div>
              <p
                style="mix-blend-mode: difference; color: rgb(255, 255, 255);"
              >
                rc watermark
              </p>
            </div>
          </div>
        </foreignobject>
      </svg>
    </div>
  </div>
</div>
`;

exports[`Watermark should render image 1`] = `
<div>
  <div
    class="rc-water-mark"
    style="z-index: 100; background-image: url(); background-size: 180px 90px;"
  >
    <div>
      <svg
        height="90"
        style="padding: 0px 0px 0px 0px; display: none;"
        viewBox="0 0 180 90"
        width="180"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <image
          height="90"
          href="data:image/png;base64,00"
          style="transform-origin: center; transform: rotate(-22deg);"
          width="180"
          x="0"
          xlink:href="data:image/png;base64,00"
          y="0"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Watermark should render image and test opacity 1`] = `
<div>
  <div
    class="rc-water-mark"
    style="z-index: 100; background-image: url(); background-size: 180px 90px;"
  >
    <div>
      <svg
        height="90"
        style="padding: 0px 0px 0px 0px; opacity: 0.2; display: none;"
        viewBox="0 0 180 90"
        width="180"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <image
          height="90"
          href="data:image/png;base64,00"
          style="transform-origin: center; transform: rotate(-22deg);"
          width="180"
          x="0"
          xlink:href="data:image/png;base64,00"
          y="0"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Watermark test false value fullPage 1`] = `
<div>
  <div
    class="rc-water-mark rc-water-mark--full"
    style="z-index: 100; background-image: url(); background-size: 100px 100px;"
  >
    <div>
      <svg
        height="100"
        style="padding: 0px 0px 0px 0px; opacity: 0.2; display: none;"
        viewBox="0 0 100 100"
        width="100"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <foreignobject
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            style="transform-origin: center; transform: rotate(-22deg);"
            xmlns="http://www.w3.org/1999/xhtml"
          >
            <span
              style="color: rgb(220, 222, 224);"
            >
              rc-ui-lib
            </span>
          </div>
        </foreignobject>
      </svg>
    </div>
  </div>
</div>
`;

exports[`Watermark test width, height, rotate, zIndex 1`] = `
<div>
  <div
    class="rc-water-mark"
    style="z-index: 999; background-image: url(); background-size: 180px 90px;"
  >
    <div>
      <svg
        height="90"
        style="padding: 0px 0px 0px 0px; display: none;"
        viewBox="0 0 180 90"
        width="180"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <foreignobject
          height="90"
          width="180"
          x="0"
          y="0"
        >
          <div
            style="transform-origin: center; transform: rotate(22deg);"
            xmlns="http://www.w3.org/1999/xhtml"
          >
            <span
              style="color: rgb(220, 222, 224);"
            >
              rc-ui-lib
            </span>
          </div>
        </foreignobject>
      </svg>
    </div>
  </div>
</div>
`;
